machine_learning_hep/fitting/roofitter.py

```diff
@@ -21,16 +23,34 @@ def __init__(self):
         ROOT.RooMsgService.instance().setSilentMode(True)
         ROOT.RooMsgService.instance().setGlobalKillBelow(ROOT.RooFit.WARNING)
 
-    def fit_mass_new(self, hist, fit_spec, roows = None, plot = False):
+    def fit_mass_new(self, hist, fit_spec, level, roows = None, plot = False):
         if hist.GetEntries() == 0:
             raise UserWarning('Cannot fit histogram with no entries')
         ws = roows or ROOT.RooWorkspace("ws")
         var_m = fit_spec.get('var', 'm')
+
+        n_signal = RooRealVar("n_signal", "Number of signal events", 100, 0, 100000000)
+        n_background = RooRealVar("n_background", "Number of background events", 100, 0, 100000000)
+
         for comp, spec in fit_spec.get('components', {}).items():
             fn = ws.factory(spec['fn'])
             if comp == 'model':
                 model = fn
         m = ws.var(var_m)
+
+        if level == "data":
+            signal_pdf = ws.pdf("sigrefl")
+            if not signal_pdf:
+                print('"sigrefl" PDF not found, using "sig" instead.')
+                signal_pdf = ws.pdf("sig")
+            if not signal_pdf:
+                print('"sig" PDF not found.')
+            background_pdf = ws.pdf("bkg")
+            model = RooAddPdf("model",
+                              "Total model",
+                              RooArgList(signal_pdf, background_pdf),
+                              RooArgList(n_signal, n_background))
+
```

Comment:
This is based on assumptions about what names are used in the DB and I am not convinced that this is the best way of dealing with this. If we decide to rely on names from the DB, we should use `sig` and `bkg` there. But I would actually prefer to pass the names of the pdfs to be used for signal and background as parameters in the function call such that the `roofitter` stays free of assumptions on naming. In addition, we should avoid `print` in parts that behave like a library and, instead, throw an exception if there is an unrecoverable error.